### PR TITLE
LPS-126645 | master

### PR DESCRIPTION
**Pull request by brittneyq**
https://issues.liferay.com/browse/LPS-126645

Hi @david-gutierrez-mesa, @Tim-Cao , @liulukiwi, can you please review this? Thank you!

### CODE REVIEW

**Comment by brittneyq** [comment]
ci:test:sf

**Comment by brittneyq** [comment]
ci:test:relevant

**Comment by Tim-Cao** [comment]
ci:stop

**Comment by Tim-Cao** [comment]
ci:stop

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 1]
```diff
@@ -2134,8 +2134,83 @@ definition {
...
+		task ("Add a web content structure with all fields") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			NavItem.gotoStructures();
+
+			WebContentStructures.addCP(
+				structureDescription = "WC Structure Description",
+				structureName = "WC Structure Name");
 
+			DataEngine.addFields(fieldNameList = "Text,Select from List,Single Selection,Multiple Selection,Grid,Date,Numeric,Image,Rich Text,Upload,Color,Web Content,Geolocation,Link to Page");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2146]: Should add the Separator field even though there is no template for it.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 2]
```diff
@@ -2134,8 +2134,83 @@ definition {
...
+
+			WebContent.addWithStructureCP(
+				structureName = "WC Structure Name",
+				templateName = "WC Template Name",
+				webContentTitle = "WC WebContent Title");
+
+			WebContent.addWithStructureCP(
+				webContentColor = "FF0D0D",
+				webContentDate = "10/10/2020",
+				webContentMultipleSelection = "Option 1",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2174]: Should add value in Grid field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 3]
```diff
@@ -2134,8 +2134,83 @@ definition {
...
+				webContentNumeric = "9",
+				webContentRichText = "This is a Rich Text field",
+				webContentSelectFromList = "Option 2",
+				webContentSingleSelection = "Option 3",
+				webContentText = "This is a Text field",
+				webContentTitle = "WC WebContent Title");
+		}
+
+		task ("Edit web content") {
+			WebContent.editCP(
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2198]: Should edit all fields

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 4]
```diff
@@ -2134,8 +2134,83 @@ definition {
...
+				webContentTitle = "WC WebContent Title",
+				webContentTitleEdit = "WC WebContent Title Edit");
+		}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2202]: Add steps to publish web content then assert edited value

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 5]
```diff
@@ -2134,8 +2134,83 @@ definition {
...
+				webContentSelectFromList = "Option 2",
+				webContentSingleSelection = "Option 3",
+				webContentText = "This is a Text field",
+				webContentTitle = "WC WebContent Title");
+		}
+
+		task ("Edit web content") {
+			WebContent.editCP(
+				webContentTitle = "WC WebContent Title",
+				webContentTitleEdit = "WC WebContent Title Edit");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2200]: Edit all field values

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 6]
```diff
@@ -2134,8 +2134,83 @@ definition {
 	test ViewAndEditAsWellAsDeleteContentWithTitleAndAllFields {
 		property portal.acceptance = "false";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2136]: Add a page and a web content for Web Content and Link to Page field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 7]
```diff
@@ -2134,8 +2134,83 @@ definition {
...
+				webContentTitle = "WC WebContent Title",
+				webContentTitleEdit = "WC WebContent Title Edit");
+		}
+
+		task ("Assert web content has been edited") {
+			WebContent.viewTitle(webContentTitle = "WC WebContent Title Edit");
+		}
+
+		task ("Delete web content") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:2208]: Unnecessary step

**Comment by david-gutierrez-mesa** [comment]
Just started reviewing :)

**Comment by david-gutierrez-mesa** [comment]
Resend here: https://github.com/david-gutierrez-mesa/liferay-portal/pull/124

**Pull request closed without merging**